File: 1072_flip_columns_for_maximum_number_of_equal_rows.cpp
class Solution {
public:
    int maxEqualRowsAfterFlips(vector<vector<int>>& arr) {
        unordered_map<string, int> hm;
        for (auto& a : arr) {
            string temp;
            for (int b : a) temp += to_string(b ^ a[0]);
            hm[temp]++;
        }
        int mx = 0;
        for (auto& val : hm) mx = max(mx, val.second);
        return mx;
    }
};

File: 1072_flip_columns_for_maximum_number_of_equal_rows.java
class Solution {
    public int maxEqualRowsAfterFlips(int[][] n) {
        var l = new HashMap<String, Integer>();
        for (var e : n) {
            var j = new StringBuilder();
            for (var q : e) j.append(q ^ e[0]);
            l.merge(j.toString(), 1, Integer::sum);
        }
        return l.values().stream().reduce(0, Math::max);
    }
}
